File: javascript/MenstrualApp.js
const prompt = require('prompt-sync')();


function calculateNextPeriod(startDay, cycleLength) {
  return startDay + cycleLength;
}

function calculateOvulation(startDay, cycleLength) {
  return startDay + (cycleLength - 14);
}

function getFertileWindow(ovulationDay) {
  return [ovulationDay - 2, ovulationDay + 2];
}

function getSafeDays(startDay, ovulationDay, nextPeriod) {
  return {
    before: [startDay, ovulationDay - 3],
    after: [ovulationDay + 3, nextPeriod]
  };
}

function getValidInput(promptText, min, max) {
  while (true) {
    const input = parseInt(prompt(promptText));
    if (!isNaN(input) && input >= min && input <= max) {
      return input;
    }
    console.log(`Please enter a number between ${min}-${max}`);
  }
}


function runTracker() {
  console.log("\nMenstrual Cycle Tracker\n");
  
  const startDay = getValidInput("Last period start day (1-31): ", 1, 31);
  const cycleLength = getValidInput("Average cycle length (21-35 days): ", 21, 35);
  
  const nextPeriod = calculateNextPeriod(startDay, cycleLength);
  const ovulationDay = calculateOvulation(startDay, cycleLength);
  const [fertileStart, fertileEnd] = getFertileWindow(ovulationDay);
  const { before, after } = getSafeDays(startDay, ovulationDay, nextPeriod);
  
  console.log("\nResults");
  console.log(`Next period: Day ${nextPeriod}`);
  console.log(`Ovulation day: Day ${ovulationDay}`);
  console.log(`Fertile window: Days ${fertileStart}-${fertileEnd}`);
  console.log(`Safe days before ovulation: ${before[0]}-${before[1]}`);
  console.log(`Safe days after ovulation: ${after[0]}-${after[1]}`);
}


runTracker();

File: Python/menstrual_app.py
def calculate_next_period(start_day: int, cycle_length: int) -> int:
    """Calculate next period date"""
    return start_day + cycle_length

def calculate_ovulation(start_day: int, cycle_length: int) -> int:
    """Calculate ovulation day (14 days before next period)"""
    return start_day + (cycle_length - 14)

def get_fertile_window(ovulation_day: int) -> str:
    """Calculate fertile window (2 days before/after ovulation)"""
    return f"{ovulation_day - 2}-{ovulation_day + 2}"

def get_safe_days(start_day: int, ovulation_day: int, next_period: int) -> str:
    """Get safe days recommendations"""
    return (
        f"Safe days before ovulation: {start_day}-{ovulation_day - 3}\n"
        f"Safe days after ovulation: {ovulation_day + 3}-{next_period}"
    )

def main():
	print("Menstrual Period Tracker")
	start_day = int(input("Enter last period start day (1-31): "))
	cycle_length = int(input("Enter average cycle length (21-35 days): "))


	next_period = calculate_next_period(start_day, cycle_length)
	ovulation_day = calculate_ovulation(start_day, cycle_length)
    
	print("\n\t Your Cycle Info")
	print(f"Next period: Day {next_period}")
	print(f"Ovulation day: Day {ovulation_day}")
	print(f"Fertile window: Days {get_fertile_window(ovulation_day)}")
	print(get_safe_days(start_day, ovulation_day, next_period))

if __name__ == "__main__":
	main()
